### PR TITLE
Add (and use) COND 

### PR DESCRIPTION
**Pull request by notmgsk**

### CODE REVIEW

File: src/library.lisp [hunk 1]
```diff
@@ -37,6 +37,15 @@
      (coalton:true ,then)
      (coalton:false ,else)))
 
+(cl:defmacro coalton::cond ((clause-a then-a) cl:&rest clauses)
+  (cl:if (cl:not (cl:null clauses))
+         `(coalton:if ,clause-a
+                      ,then-a
+                      (coalton::cond ,@clauses))
+         (cl:progn
+           (cl:assert (cl:eql cl:t clause-a) () "COND must have a catch-all T clause.")
```
Comment (notmgsk) [inline src/library.lisp]: `cl:t` or `otherwise`?

Comment (notmgsk) [reply src/library.lisp]: or `else`?

File: src/library.lisp [hunk 2]
```diff
@@ -37,6 +37,15 @@
      (coalton:true ,then)
      (coalton:false ,else)))
 
+(cl:defmacro coalton::cond ((clause-a then-a) cl:&rest clauses)
```
Comment (stylewarning) [inline src/library.lisp]:
```suggestion
(cl:defmacro coalton:cond ((clause-a then-a) cl:&rest clauses)
```

File: src/library.lisp [hunk 3]
```diff
@@ -37,6 +37,15 @@
...
+  (cl:if (cl:not (cl:null clauses))
```
Comment (stylewarning) [inline src/library.lisp]:
```suggestion
  (cl:if (cl:not (cl:endp clauses))
```

File: src/library.lisp [hunk 4]
```diff
@@ -37,6 +37,15 @@
...
+         `(coalton:if ,clause-a
+                      ,then-a
+                      (coalton::cond ,@clauses))
```
Comment (stylewarning) [inline src/library.lisp]:
```suggestion
                      (coalton:cond ,@clauses))
```

File: src/library.lisp [hunk 5]
```diff
@@ -37,6 +37,15 @@
...
+         (cl:progn
+           (cl:assert (cl:eql 'else clause-a) () "COND must have a catch-all T clause.")
```
Comment (stylewarning) [inline src/library.lisp]:
```suggestion
           (cl:assert (cl:eq 'coalton:else clause-a) () "COND must have an ELSE clause.")
```

File: src/package.lisp [hunk 6]
```diff
@@ -26,6 +26,7 @@
    #:let
    #:letrec
    #:if
+   #:cond
```
Comment (stylewarning) [inline src/package.lisp:29]:
```suggestion
   #:cond
   #:else          ; syntax for COND
```

File: src/library.lisp [hunk 7]
```diff
@@ -132,21 +141,15 @@
...
-                            (gcd (half (- v u)) u)))))))))
+    (cond ((= u v)   u)
+          ((zerop u) v)
+          ((zerop v) u)
+          ((evenp u) (if (evenp v)
+                         (double (gcd (half u) (half v)))
+                         (gcd (half u) v)))
+          (else      (cond ((evenp v) (gcd u (half v)))
+                           ((> u v)   (gcd (half (- u v)) v))
+                           (else      (gcd (half (- v u)) u)))))))
```
Comment (stylewarning) [inline src/library.lisp:152]: Beautiful!

**Review by stylewarning — CHANGES_REQUESTED** [review]

**Comment by notmgsk** [comment]
@stylewarning I added a couple more changes that might be controversial. Namely, I removed the `coalton:` prefixes in places where they're not technically required, and I capitalised the symbols `false` and `true`. I can revert those if you think it's better to be clear about where those symbols are coming from.

File: src/library.lisp [hunk 8]
```diff
@@ -34,12 +34,21 @@
 
 (cl:defmacro coalton:if (expr then else)
   `(match ,expr
-     (coalton:true ,then)
```
Comment (stylewarning) [inline src/library.lisp:37]: Removal here is OK. I mostly want to keep the prefixes at definitions.

File: src/library.lisp [hunk 9]
```diff
@@ -34,12 +34,21 @@
...
+              ,then-a
+              (cond ,@clauses))
+         (cl:progn
+           (cl:assert (cl:eq 'else clause-a) () "COND must have an ELSE clause.")
+           then-a)))
 
 (cl:declaim (cl:inline lisp-boolean-to-coalton-boolean))
 (cl:defun lisp-boolean-to-coalton-boolean (x)
-  (cl:if x coalton:true coalton:false))
+  (cl:if x True False))
```
Comment (stylewarning) [inline src/library.lisp:51]: good change, I prefer this

**Review by stylewarning — APPROVED** [review]

**Pull request merged**